stripe/src/main/java/com/stripe/android/StripeApiHandler.java

```diff
@@ -612,6 +627,66 @@ static Customer retrieveCustomer(@NonNull String customerId, @NonNull String sec
         return Customer.fromString(response.getResponseBody());
     }
 
+    @Nullable
+    static String retrieveIssuingCardPin(
+            @NonNull String cardId,
+            @NonNull String verificationId,
+            @NonNull String userOneTimeCode,
+            @NonNull String ephemeralKeySecret)
+            throws InvalidRequestException,
+            APIConnectionException,
+            APIException,
+            AuthenticationException,
+            CardException, JSONException {
+        Map<String, Object> paramsMap = new HashMap<>();
+
+        Map<String, Object> verificationMap = new HashMap<>();
+        verificationMap.put("id", verificationId);
+        verificationMap.put("one_time_code", userOneTimeCode);
+
+        paramsMap.put("verification", verificationMap);
+
+        StripeResponse response = getStripeResponse(
+                GET,
+                getRetrieveIssuingCardPinUrl(cardId),
+                paramsMap,
+                RequestOptions.builder(ephemeralKeySecret).setApiVersion(API_VERSION).build());
+        // Method throws if errors are found, so no return value occurs.
+        convertErrorsToExceptionsAndThrowIfNecessary(response);
+        JSONObject jsonResponse = new JSONObject(response.getResponseBody());
+        return (String) jsonResponse.get("pin");
+    }
+
+    @Nullable
+    static void updateIssuingCardPin(
+            @NonNull String cardId,
+            @NonNull String newPin,
+            @NonNull String verificationId,
+            @NonNull String userOneTimeCode,
+            @NonNull String ephemeralKeySecret)
+            throws InvalidRequestException,
+            APIConnectionException,
+            APIException,
+            AuthenticationException,
+            CardException {
+        Map<String, Object> paramsMap = new HashMap<>();
+
+        Map<String, Object> verificationMap = new HashMap<>();
+        verificationMap.put("id", verificationId);
+        verificationMap.put("one_time_code", userOneTimeCode);
+
+        paramsMap.put("verification", verificationMap);
```

Comment:
what about creating a `createVerificationParams()` method to avoid redundant hard coded strings (here and line 647)?

Reply:
👍